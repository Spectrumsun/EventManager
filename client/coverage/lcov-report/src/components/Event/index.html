<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Event</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Event
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">17.11% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>13/76</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/37</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/31</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.84% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>13/69</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Addevent.js"><a href="Addevent.js.html">Addevent.js</a></td>
	<td data-value="10" class="pic low"><div class="chart"><div class="cover-fill" style="width: 10%;"></div><div class="cover-empty" style="width:90%;"></div></div></td>
	<td data-value="10" class="pct low">10%</td>
	<td data-value="20" class="abs low">2/20</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="10.53" class="pct low">10.53%</td>
	<td data-value="19" class="abs low">2/19</td>
	</tr>

<tr>
	<td class="file low" data-value="EditEvent.js"><a href="EditEvent.js.html">EditEvent.js</a></td>
	<td data-value="12.5" class="pic low"><div class="chart"><div class="cover-fill" style="width: 12%;"></div><div class="cover-empty" style="width:88%;"></div></div></td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="24" class="abs low">3/24</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="13.64" class="pct low">13.64%</td>
	<td data-value="22" class="abs low">3/22</td>
	</tr>

<tr>
	<td class="file low" data-value="EventInfo.js"><a href="EventInfo.js.html">EventInfo.js</a></td>
	<td data-value="16.67" class="pic low"><div class="chart"><div class="cover-fill" style="width: 16%;"></div><div class="cover-empty" style="width:84%;"></div></div></td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="18" class="abs low">3/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="13" class="abs low">0/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="18.75" class="pct low">18.75%</td>
	<td data-value="16" class="abs low">3/16</td>
	</tr>

<tr>
	<td class="file low" data-value="ViewEvents.js"><a href="ViewEvents.js.html">ViewEvents.js</a></td>
	<td data-value="27.27" class="pic low"><div class="chart"><div class="cover-fill" style="width: 27%;"></div><div class="cover-empty" style="width:73%;"></div></div></td>
	<td data-value="27.27" class="pct low">27.27%</td>
	<td data-value="11" class="abs low">3/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="9" class="abs low">3/9</td>
	</tr>

<tr>
	<td class="file medium" data-value="getEvent.js"><a href="getEvent.js.html">getEvent.js</a></td>
	<td data-value="66.67" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 66%;"></div><div class="cover-empty" style="width:34%;"></div></div></td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Apr 15 2018 00:07:57 GMT+0100 (WAT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
