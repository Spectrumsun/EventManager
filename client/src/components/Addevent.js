import React from 'react';
import EventForm from './EventForm';

const eventCenter = () => (
  <div className="container" style={{ paddingTop: '100px' }}>
    <div className="card card w-50 loginCard">
      <div className="card-header dark">
        <h1 className="color">Add Event</h1>
      </div>
      <EventForm />
    </div>
  </div>
);

export default eventCenter;
