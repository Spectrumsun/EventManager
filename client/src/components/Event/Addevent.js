import React, { Component } from 'react';
import { connect } from 'react-redux';
import toast from 'toastr';
import PropTypes from 'prop-types';
import EventForm from './Form/EventForm';
import * as action from '../../store/actions/index';

/**
 * @class AddEvent
 *
 * @extends {React.Component}
 */
class AddEvent extends Component {
   state = {
     name: '',
     date: '',
     time: '',
     purpose: '',
     center: ''
   }

   /**
   * @description update component state with current value in dom
   *
   * @param {any} event
   *
   * @memberof AddEvent
   *
   * @returns {void}
   */
  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }


  /**
   * @description vaildate data in state
   * sends state to api with action dispatch
   * @param {any} event
   *
   * @memberof AddEvent
   *
   * @returns {void}
   */
  onSubmit = (e) => {
    e.preventDefault();
    if (this.state.name === '') {
      toast.error('Event Name cannot be blank');
    } else if (this.state.date === '') {
      toast.error('Event Date cannot be blank');
    } else if (this.state.time === '') {
      toast.error('Event Time cannot be blank');
    } else if (this.state.purpose === '') {
      toast.error('Event Purpose cannot be blank');
    } else if (this.state.center === '') {
      toast.error('You have to choose a Center');
    } else {
      this.props.initPostEvent(
        this.state,
        this.props.history
      );
    }
  }

  /**
   * @description get selectCenter id and state state
   * @param {any} event
   *
   * @memberof AddEvent
   *
   * @returns {void}
   */
  selectCenter = (id) => {
    this.state.center = id;
  }

  /**
   * @description renders component to the DOM
   *
   * @memberof AddEvent
   *
   * @returns {JSX} JSX representation of component
   */
  render() {
    return (
      <div className="container" style={{ paddingTop: '100px' }}>
        <div className="card card w-50 loginCard">
          <div className="card-header dark">
            <h1 className="color">Add Event</h1>
          </div>
          <EventForm
            onChange={this.onChange}
            onSubmit={this.onSubmit}
            name={this.state.name}
            date={this.state.date}
            time={this.state.time}
            purpose={this.state.purpose}
            selectCenter={id => this.selectCenter(id)}
          />
        </div>
      </div>
    );
  }
}

AddEvent.propTypes = {
  initPostEvent: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired,
};


const mapDispatchToProps = dispatch => ({
  initPostEvent: (input, history) =>
    dispatch(action.initPostEvent(input, history)),
});


export default connect(null, mapDispatchToProps)(AddEvent);

