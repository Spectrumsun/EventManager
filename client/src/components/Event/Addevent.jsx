import React, { Component } from 'react';
import { connect } from 'react-redux';
import toast from 'toastr';
import PropTypes from 'prop-types';
import EventForm from './Form/EventForm';
import * as action from '../../store/actions/index';

/**
 * @class AddEvent
 *
 * @extends {React.Component}
 */
export class AddEvent extends Component {
   state = {
     name: '',
     startDate: '',
     endDate: '',
     time: '',
     purpose: '',
     center: '',
     totalPage: '',
     next: 1,
     centerName: '',
     pageNumber: '',
     formValid: false,
   }


   /**
   * @description update component state with current value in dom
   *
   * @param {any} event
   *
   * @memberof AddEvent
   *
   * @returns {void}
   */
  onChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }


  /**
   * @description vaildate data in state
   * sends state to api with action dispatch
   * @param {any} event
   *
   * @memberof AddEvent
   *
   * @returns {void}
   */
  onSubmit = (event) => {
    event.preventDefault();
    this.setState({ formValid: false });
    const reWhiteSpace = new RegExp(/^\s+$/);
    if (this.state.name === '') {
      toast.error('Event Name cannot be blank');
    } else if (reWhiteSpace.test(this.state.fullname) === true) {
      toast.error('Event Name cannot white space');
    } else if (this.state.startDate === '') {
      toast.error('Event start date cannot be blank');
    } else if (this.state.endDate === '') {
      toast.error('Event end date cannot be blank');
    } else if (this.state.endDate < this.state.startDate) {
      toast.error('!Event end date cannot be behind event start date');
    } else if (this.state.time === '') {
      toast.error('Event Time cannot be blank');
    } else if (this.state.purpose === '') {
      toast.error('Event Purpose cannot be blank');
    } else if (this.state.center === '') {
      toast.error('You have to choose a Center');
    } else {
      this.setState({ formValid: true });
      this.props.initPostEvent(
        this.state,
        this.props.history
      ).then(() => {
        this.setState({ formValid: false });
      });
    }
  }

  getCenter = () => {
    this.props.onInitCenters(3, 1);
  }


  add = () => {
    this.setState({ totalPage: this.props.page.pages });
    this.state.totalPage = this.props.page.pages;
    if (this.state.next < this.state.totalPage) {
      const nextPage = ++this.state.next;
      this.state.pageNumber = nextPage;
      this.setState({ pageNumber: nextPage });
      this.setState({ next: nextPage });
      this.props.onInitCenters(3, nextPage);
    }
  }


  minus = () => {
    const limit = 1;
    if (limit < this.state.next) {
      const nextPage = --this.state.next;
      this.state.pageNumber = nextPage;
      this.setState({ pageNumber: nextPage });
      this.props.onInitCenters(3, nextPage);
    }
  }

  /**
   * @description get selectCenter id and state state
   * @param {any} event
   *
   * @memberof AddEvent
   *
   * @returns {void}
   */
  selectCenter = (id, myCenter) => {
    this.state.center = id;
    this.setState({ centerName: myCenter });
  }

  /**
   * @description renders component to the DOM
   *
   * @memberof AddEvent
   *
   * @returns {JSX} JSX representation of component
   */
  render() {
    const numberOfPages = (
      <li className="page-item">
        <a className="page-link">
                Page {this.state.pageNumber} of {this.state.totalPage}
        </a>
      </li>);
    const numberOfPages1 = (
      <li className="page-item">
        <a className="page-link">
                  Page 1  of {this.props.page.pages }
        </a>
      </li>);
    const showCenterNane = (
      <h6 className="list-group-item col-md-9">
        Center Name : {this.state.centerName}
      </h6>);

    return (
      <div className="container" style={{ paddingTop: '100px' }}>
        <div className="card card w-50 loginCard">
          <div className="card-header dark">
            <h1 className="color">Add Event</h1>
          </div>
          <EventForm
            onChange={this.onChange}
            onSubmit={this.onSubmit}
            name={this.state.name}
            startDate={this.state.startDate}
            endDate={this.state.endDate}
            time={this.state.time}
            purpose={this.state.purpose}
            getCenter={this.getCenter}
            selectCenter={(id, myCenter) =>
              this.selectCenter(id, myCenter)}
            add={this.add}
            minus={this.minus}
            centerName={this.state.centerName}
            numberOfPages={numberOfPages}
            totalPage={this.state.totalPage}
            numberOfPages1={numberOfPages1}
            showCenterNane={showCenterNane}
            formValid={this.state.formValid}
          />
        </div>
      </div>
    );
  }
}

AddEvent.propTypes = {
  initPostEvent: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired,
  onInitCenters: PropTypes.func.isRequired,
  page: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  page: state.centers.pagination,
});


const mapDispatchToProps = dispatch => ({
  initPostEvent: (input, history) =>
    dispatch(action.initPostEvent(input, history)),
  onInitCenters: (limit, page) =>
    dispatch(action.initCenters(limit, page))
});


export default connect(mapStateToProps, mapDispatchToProps)(AddEvent);

