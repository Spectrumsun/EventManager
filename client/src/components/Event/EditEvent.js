import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import toast from 'toastr';
import EventForm from './Form/EventForm';
import * as action from '../../store/actions/index';

/**
 * @class EditEvent
 *
 * @extends {React.Component}
 */
class EditEvent extends Component {
   state = {
     name: this.props.loadedEvent.eventName,
     date: this.props.loadedEvent.eventdate,
     time: this.props.loadedEvent.time,
     purpose: this.props.loadedEvent.purpose,
     center: this.props.loadedEvent.centerId
   }

  /**
   * @description run action on component mount to reload data
   *
   * @param {any} props.params.token
   *
   * @memberof EditEvent
   */
   componentWillMount() {
     this.props.onOneEvent(this.props.match.params.id);
   }

  /**
   * @description update component state with current value in dom
   *
   * @param {any} event
   *
   * @memberof EditEvent
   *
   * @returns {void}
   */
  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }


  /**
   * @description vaildate data in state
   * sends state to api with action dispatch
   * @param {any} event
   *
   * @memberof EditEvent
   *
   * @returns {void}
   */
   onSubmit = (e) => {
     e.preventDefault();
     if (this.state.name === '') {
       toast.error('Event Name cannot be blank');
     } else if (this.state.date === '') {
       toast.error('Event Date cannot be blank');
     } else if (this.state.time === '') {
       toast.error('Event Time cannot be blank');
     } else if (this.state.purpose === '') {
       toast.error('Event Purpose cannot be blank');
     } else if (this.state.center === '') {
       toast.error('You have to choose a Center');
     } else {
       this.props.initEditEvent(
         this.props.match.params.id, this.state,
         this.props.history
       );
     }
   }
   /**
   * @description get selectCenter id and state state
   * @param {any} event
   *
   * @memberof EditEvent
   *
   * @returns {void}
   */
   selectCenter = (id) => {
     this.state.center = id;
   }

   /**
   * @description renders component to the DOM
   *
   * @memberof EditEvent
   *
   * @returns {JSX} JSX representation of component
   */
   render() {
     return (
       <div className="container" style={{ paddingTop: '100px' }}>
         <div className="card card w-50 loginCard">
           <div className="card-header dark">
             <h1 className="color">Edit Event</h1>
           </div>
           <EventForm
             onChange={this.onChange}
             onSubmit={this.onSubmit}
             name={this.state.name}
             date={this.state.date}
             time={this.state.time}
             purpose={this.state.purpose}
             selectCenter={id => this.selectCenter(id)}
           />
         </div>
       </div>
     );
   }
}

// EditEvent.propTypes = {
//   onOneEvent: PropTypes.func.isRequired,
//   initEditEvent: PropTypes.func.isRequired,
//   loadedEvent: PropTypes.shape({
//     eventName: PropTypes.string,
//     eventdate: PropTypes.string,
//     purpose: PropTypes.string,
//     time: PropTypes.string,
//     userId: PropTypes.number,
//     centerId: PropTypes.number,
//   }).isRequired,
//   history: PropTypes.shape({}).isRequired,
//   match: PropTypes.shape({
//     params: PropTypes.shape({
//       id: PropTypes.string,
//     }),
//   }).isRequired,
// };


const mapStateToProps = state => ({
  loadedEvent: state.events.loadedEvent,
  error: state.events.error
});

const mapDispatchToProps = dispatch => ({
  onOneEvent: id => dispatch(action.initGetOneEvent(id)),
  initEditEvent: (id, events, history) =>
    dispatch(action.initEditEvent(id, events, history)),
});


export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditEvent);

