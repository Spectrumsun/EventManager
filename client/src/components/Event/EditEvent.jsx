import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import EventForm from './Form/EventForm';
import * as action from '../../store/actions/index';
import { checkEvent } from '../../static/js/validator';

/**
 * @class EditEvent
 *
 * @extends {React.Component}
 */
export class EditEvent extends Component {
   state = {
     name: this.props.loadedEvent.eventName,
     startDate: this.props.loadedEvent.startDate,
     endDate: this.props.loadedEvent.endDate,
     time: this.props.loadedEvent.time,
     purpose: this.props.loadedEvent.purpose,
     center: this.props.loadedEvent.centerId,
     totalPage: '',
     next: 1,
     centerName: '',
     pageNumber: '',
     formValid: false,
   }

  /**
   * @description run action on component mount to reload data
   *
   * @param {any} props.params.token
   *
   * @memberof EditEvent
   */
   componentWillMount() {
     this.props.onOneEvent(this.props.match.params.id);
   }

  /**
   * @description update component state with current value in dom
   *
   * @param {any} event
   *
   * @memberof EditEvent
   *
   * @returns {void}
   */
  onChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }


  /**
   * @description vaildate data in state
   * sends state to api with action dispatch
   * @param {any} event
   *
   * @memberof EditEvent
   *
   * @returns {void}
   */
   onSubmit = (event) => {
     event.preventDefault();
     event.preventDefault();
     checkEvent(this.state.name, this.state.startDate, this.state.endDate, this.state.time, this.state.purpose, this.state.center, (err, res) => {
       if (res) {
         this.setState({ formValid: true });
         this.props.initEditEvent(
           this.props.match.params.id, this.state,
           this.props.history
         ).then(() =>
           this.setState({ formValid: false }));
       }
     });
   }


   getCenter = () => {
     this.props.onInitCenters(3, 1);
   }

   add = () => {
     this.state.totalPage = this.props.page.pages;
     if (this.state.next < this.state.totalPage) {
       const nextPage = ++this.state.next;
       this.setState({ pageNumber: nextPage });
       this.setState({ next: nextPage });
       this.props.onInitCenters(3, nextPage);
     }
   }


  minus = () => {
    const limit = 1;
    if (limit < this.state.next) {
      const nextPage = --this.state.next;
      this.state.pageNumber = nextPage;
      this.props.onInitCenters(3, nextPage);
    }
  }

  /**
   * @description get selectCenter id and state state
   * @param {any} event
   *
   * @memberof EditEvent
   *
   * @returns {void}
   */
  selectCenter = (id, myCenter) => {
    this.state.center = id;
    this.setState({ centerName: myCenter });
  }

  /**
   * @description renders component to the DOM
   *
   * @memberof EditEvent
   *
   * @returns {JSX} JSX representation of component
   */
  render() {
    const numberOfPages = (
      <li className="page-item">
        <a className="page-link">
                Page {this.state.pageNumber} of {this.state.totalPage}
        </a>
      </li>);
    const numberOfPages1 = (
      <li className="page-item">
        <a className="page-link">
                  Page 1  of {this.props.page.pages }
        </a>
      </li>);
    const showCenterNane = (
      <h6 className="list-group-item col-md-9">
        Center Name : {this.state.centerName}
      </h6>);
    return (
      <div className="container" style={{ paddingTop: '100px' }}>
        <div className="card card w-50 loginCard">
          <div className="card-header dark">
            <h1 className="color">Edit Event</h1>
          </div>
          <EventForm
            onChange={this.onChange}
            onSubmit={this.onSubmit}
            name={this.state.name}
            startDate={this.state.startDate}
            endDate={this.state.endDate}
            time={this.state.time}
            purpose={this.state.purpose}
            getCenter={this.getCenter}
            selectCenter={(id, myCenter) => this.selectCenter(id, myCenter)}
            add={this.add}
            minus={this.minus}
            centerName={this.state.centerName}
            numberOfPages={numberOfPages}
            totalPage={this.state.totalPage}
            numberOfPages1={numberOfPages1}
            showCenterNane={showCenterNane}
            formValid={this.state.formValid}
          />
        </div>
      </div>
    );
  }
}

EditEvent.propTypes = {
  onOneEvent: PropTypes.func.isRequired,
  initEditEvent: PropTypes.func.isRequired,
  loadedEvent: PropTypes.shape({
    eventName: PropTypes.string,
    startDate: PropTypes.string,
    endDate: PropTypes.endDate,
    purpose: PropTypes.string,
    time: PropTypes.string,
    userId: PropTypes.number,
    centerId: PropTypes.number,
  }).isRequired,
  history: PropTypes.shape({}).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string,
    }),
  }).isRequired,
  onInitCenters: PropTypes.func.isRequired,
};


const mapStateToProps = state => ({
  loadedEvent: state.events.loadedEvent,
  page: state.centers.pagination,
  error: state.events.error
});

export const mapDispatchToProps = dispatch => ({
  onOneEvent: id => dispatch(action.initGetOneEvent(id)),
  initEditEvent: (id, events, history) =>
    dispatch(action.initEditEvent(id, events, history)),
  onInitCenters: (limit, page) =>
    dispatch(action.initCenters(limit, page))
});


export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditEvent);

