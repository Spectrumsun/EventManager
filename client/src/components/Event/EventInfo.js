import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import * as actions from '../../store/actions/index';

class EventInfo extends Component {
  componentDidMount() {
    this.props.onOneEvent(this.props.match.params.id);
  }

  deletePost = () => {
    e.preventDefault();
    this.props.onDeleteEvent(this.props.match.params.id);
  }


  render() {
    const center = new Object(this.props.events && this.props.events.centers && this.props.events.centers);
    console.log(center);
    // console.log(events);
    const event = (
      <div className="container" style={{ paddingTop: '100px' }}>
        <div className="container" style={{ width: '45rem' }}>
          <div className="card loginCard">
            <div className="card-header dark">
              <h1 className="color">Event Info</h1>
            </div>
            <div className="card-body">
              <h5 ><strong>Event Name</strong></h5>
              <h6 className="list-group-item centerlist">{this.props.events.eventName}</h6>

              <br />

              <h5><strong>Date</strong></h5>
              <h6 className="list-group-item centerlist">{this.props.events.eventdate}</h6>
              <br />

              <h5><strong>Time</strong></h5>
              <h6 className="list-group-item centerlist">{this.props.events.time}</h6>
              <br />

              <h5><strong>Purpose</strong></h5>
              <h6 className="list-group-item centerlist">{this.props.events.purpose}</h6>
              <br />

              <h5><strong>Center</strong></h5>
              <ul className="list-group col-md-6">
                <Link to={`/centers/${center.id}`} key={center.id} style={{ color: '#35434A' }}>
                  <h6><li className="list-group-item centerlist">{center.centerName}</li></h6>
                </Link>
              </ul>
              <br />

              <Link to={`/events/edit/${this.props.match.params.id}`} key={this.props.match.params.id} style={{ color: '#35434A' }}>
                <button type="submit" className="btn btn-dark" style={{ float: 'left' }}>Edit</button>
              </Link>
              <button
                type="button"
                className="btn btn-danger"
                style={{ marginLeft: '20px' }}
                data-toggle="modal"
                data-target="#exampleModal"
              >
                  Delete
              </button>
              <div className="modal fade" id="exampleModal" tabIndex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
                <div className="modal-dialog" role="document">
                  <div className="modal-content">
                    <div className="modal-header">
                      <h5 className="modal-title" id="exampleModalLabel">Modal title</h5>
                      <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                        <span aria-hidden="true">&times;</span>
                      </button>
                    </div>
                    <div className="modal-body">
                      <h6>Are you sure you want to delete <strong>{this.props.events.eventName} ?</strong></h6>
                    </div>
                    <div className="modal-footer">
                      <button type="button" className="btn btn-secondary" data-dismiss="modal">Cancel</button>
                      <button type="button" className="btn btn-danger">Yes</button>
                    </div>
                  </div>
                </div>
              </div>


            </div>
          </div>
        </div>
      </div>
    );
    return (
      <div>
        {event}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  events: state.events.loadedEvent
});

const mapDispatchToProps = dispatch => ({
  onOneEvent: id => dispatch(actions.initGetOneEvent(id)),
  onDeleteEvent: id => dispatch(actions.initDeleteEvent(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EventInfo);
