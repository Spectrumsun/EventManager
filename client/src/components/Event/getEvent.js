import React from 'react';
import PropTypes from 'prop-types';


const display = props => (
  <div className="card d-lg-inline-block" style={{ width: '14rem' }}>
    <img
      className="card-img-top"
      src="https://res.cloudinary.com/skybound/image/upload/v1521281912/image1.jpg"
      style={{ objectFit: 'cover', height: '15rem' }}
      alt="Card cap"
    />
    <div className="card-body">
      <h4 className="card-title">{props.eventName}</h4>
      <p className="card-text">{props.eventdate}</p>
      <h6 className="card-text">{props.purpose}</h6>
      <h6 className="btn btn-primary">View more</h6>
    </div>
  </div>
);

display.propTypes = {
  eventName: PropTypes.string.isRequired,
  eventdate: PropTypes.string.isRequired,
  purpose: PropTypes.string.isRequired,
};

export default display;

