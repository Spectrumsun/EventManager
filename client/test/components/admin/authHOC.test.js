import React from 'react';
import { shallow } from 'enzyme';
import Authenticate from '../../../src/components/Auth/authHOC';
import { AddEvent } from '../../../src/components/Event/Addevent'

const components = Authenticate(AddEvent);

const { WrappedComponent } = components;


let props;
let mountedComponent;

/**
 * @description Initialise the component
 *
 * @returns {object} ManageRecipe - Mounted component
 */
const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<WrappedComponent {...props} />);
  }
  return mountedComponent;
};

/**
 * @description Initialise the component
 *
 * @returns {object} ManageRecipe - Mounted component
 */

describe('Component: Authenticateenticate Admin', () => {
  beforeEach(() => {
    props = {
      isAuthenticated: false,
      auth: {
        isAuthenticated: false,
        user: null
      },
      history: {
        push: jest.fn()
      }
    };
    mountedComponent = undefined;
  });

  describe('Protected Routes test suite', () => {
    it('Redirects user to homepage if they are not Admin', () => {
      props.isAuthenticated = false;
      expect(getComponent()).toMatchSnapshot();
    });
  });
});

