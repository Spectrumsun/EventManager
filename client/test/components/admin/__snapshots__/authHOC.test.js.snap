// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Authenticateenticate Admin Protected Routes test suite Redirects user to homepage they are not Admin 1`] = `
<AddEvent
  auth={
    Object {
      "isAuthenticated": false,
      "user": null,
    }
  }
  history={
    Object {
      "push": [MockFunction] {
        "calls": Array [
          Array [
            "/login",
          ],
        ],
      },
    }
  }
  isAuthenticated={false}
/>
`;
